--[[HANDGUN v1.2 CREATED BY DESTINYHARBINGER
	
A fully working, non virus riddled, and decently realistic up to date handgun made from a Unioned Part
made from solid modeling. This weapon is actually an edited Template Assualt Rifle you can see from those
broken template games. Modified to become more modern and up to date.

This weapon features:
-Anti Teamkill
-Flashes and Flares and Bears Oh My!
-Reloading (WOWWWWWWW)
-IOS and Android Compatibility (Tested and Approved)
-No virus junk.
-Animations (will improve later)
-Easy to edit script!
-Decent pistol sounds!
-Non-laggy! Only 2 parts and thats it!
-and not so much more because its just a regular gun!!!!!!!

)))))))))IMPORTANT:
DO NOT DELETE, UNUNION, OR CHANGE NAMES OF ANYTHING UNLESS YOU ARE AN EXPERT AT DOING THAT BUISNESS.
TREAD CAREFULLY.


~PATCH NOTES~
v1.2
-Tool not working if in starterpack has been fixed by Luckymaxer.
-Minor HUD changes.


v1.1
-Smoke is now disabled. Go through the handgun script and CTRL + F, search smoke, and uncomment the options
for it to work again.
-Shell casings have been added
-When sucessfully hitting someone, a local sound is played to indicate a hit.
-Works with mobile now. Thats right, you can now use it on iPhone Sean.
-Changed Equipping sounds.

______  _____  _____  _____  _____  _   _ __   __              
|  _  \|  ___|/  ___||_   _||_   _|| \ | |\ \ / /              
| | | || |__  \ `--.   | |    | |  |  \| | \ V /               
| | | ||  __|  `--. \  | |    | |  | . ` |  \ /                
| |/ / | |___ /\__/ /  | |   _| |_ | |\  |  | |                
|___/  \____/ \____/   \_/   \___/ \_| \_/  \_/                
                                                               
                                                               
 _   _   ___  ______ ______  _____  _   _  _____  _____ ______ 
| | | | / _ \ | ___ \| ___ \|_   _|| \ | ||  __ \|  ___|| ___ \
| |_| |/ /_\ \| |_/ /| |_/ /  | |  |  \| || |  \/| |__  | |_/ /
|  _  ||  _  ||    / | ___ \  | |  | . ` || | __ |  __| |    / 
| | | || | | || |\ \ | |_/ / _| |_ | |\  || |_\ \| |___ | |\ \ 
\_| |_/\_| |_/\_| \_|\____/  \___/ \_| \_/ \____/\____/ \_| \_|
  __    __    _____  _____  _____  ______      _____  _____  __   _____ 
 / /    \ \  / __  \|  _  ||  _  ||___  /     / __  \|  _  |/  | |  ___|
| |  ___ | | `' / /'| |/' || |/' |   / /______`' / /'| |/' |`| | |___ \ 
| | / __|| |   / /  |  /| ||  /| |  / /|______| / /  |  /| | | |     \ \
| || (__ | | ./ /___\ |_/ /\ |_/ /./ /        ./ /___\ |_/ /_| |_/\__/ /
| | \___|| | \_____/ \___/  \___/ \_/         \_____/ \___/ \___/\____/ 
 \_\    /_/     

If you can, leave me some credit k ;)

]]